docs/RFCS/20222202_historical_hot_ranges.md

```diff
@@ -0,0 +1,460 @@
+#### Tenant Sampling 
+Tenant sampling generates metrics for a specific tenant. Tenant
+sampling is performed by the host, and metrics are stored such that a tenant can
+only access metrics for ranges inside its own keyspace. This prevents a tenant from
+accessing hotness from system-level ranges or another tenant's ranges. Every
+tenant receives its own 1000 range budget to prevent colder tenants from being
+overshadowed by hotter tenants.
+
+#### Host Sampling 
+Host sampling generates metrics reported over the cluster's
+entire keyspace in a tenant-agnostic way. The host receives its own 1000 range
+budget.
+
+A dedicated single-tenant deployment only performs host sampling, while
+multi-tenant deployments perform both tenant and host sampling. In a
+single-tenant deployment, the results of tenant's sampling sampling would be
+contained in the host sampling, and therefore be redundant.
+
+Tenant sampling and Host sampling follow mostly the same process - the
+differences manifest in how aggregation is performed to keep samples within
+budget, and how the sampled metrics are written to `system.hot_ranges`.
+
+According to the sample rate, a job is executed to
+request the 128 hottest replicas from every store via `replRankings`. These hot
+replicas are then sorted alphabetically by their start keys. If tenant sampling
+is being performed, each set of a tenant's reported replicas is
+limited to 1000 by a range reduction and aggregation scheme. This range
+reduction scheme is discussed in the next section. If host sampling is being
+performed, the global set of reported replicas is limited to 1000 by the same
+range reduction scheme. 
+
+After ranges have been aggregated and reduced, they are written to
+`system.hot_ranges`. Tenant sampling creates a row per tenant per sample, while
+Host sampling creates a single row per sample, whose `tenant_id` value is set to
+`'host'`. To query `system.hot_ranges`, demateralized views are created for each
+tenant and for the host.
+
+Finally, existing samples older than 2 weeks will be deleted.
+
+
+
+```sql 
+-- schema for system.hot_ranges
+CREATE TABLE system.hot_ranges ( 
+    sample_time INT NOT NULL,
+    
+    -- When host sampling is being performed, 
+    -- tenant_id will be set to 'host'
+    tenant_id INT NOT NULL,
+
+    -- info is defined below by `HHRSample`
+    info BYTES NOT NULL,
+
+    PRIMARY KEY (sample_time, tenant_id)
+); 
+```
+
+```protobuf
+// protobuf definition for HHRSample
+message HHRSample { 
+    message HotRange {
+        message Schema { 
+            string database = 1;
+            repeated string tables = 2; 
+        } 
+        int32 range_id = 1; 
+        int32 replica_id = 2; 
+        float qps = 3; 
+        string start_key = 4; 
+        string end_key = 5; 
+        repeated int32 node_ids = 6;
+        repeated int32 store_ids = 7; 
+        repeated Schema schema = 8; 
+        string locality = 9; 
```

Comment:
can you say more about what this is?

Reply:
This -  https://github.com/cockroachdb/cockroach/blob/master/docs/generated/http/full.md#locality-5

Seeing how `Locality` isn't a string, I could stringify the value, or adjust the proto.